library/Notifications/Widget/Timeline/EntryFlyout.php

```diff
@@ -251,9 +291,24 @@ protected function generateAndSetRotationInfo(): static
 
         $noneType = \IntlDateFormatter::NONE;
         $shortType = \IntlDateFormatter::SHORT;
-        $timeFormatter = new \IntlDateFormatter(\Locale::getDefault(), $noneType, $shortType);
-        $dateFormatter = new \IntlDateFormatter(\Locale::getDefault(), $shortType, $noneType);
-        $firstHandoff = $dateFormatter->format(DateTime::createFromFormat('Y-m-d', $this->firstHandoff));
+        $scheduleTimezone = new DateTimeZone(ScheduleTimezoneStorage::getScheduleTimezone());
+        $displayTimezone = new DateTimeZone((new TimezonePicker($scheduleTimezone->getName()))->getDisplayTimezone());
```

Comment:
You don't need `ScheduleTimezoneStorage::getScheduleTimezone()`, when an `Timeline\Entry` is created, it's done by the rotation which knows its own tz very well, so just teach such entries about their own tz. It might be a good idea to change `EntryFlyout::withActiveMember()` to `EntryFlyout::forEntry()`.

Similarly, the display tz should be a mandatory requirement for a flyout. Make it a constructor argument. The timeline, which is responsible to prepare and construct flyouts, knows the display tz very well and can pass it on.